<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1101</width>
    <height>920</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color: rgb(255, 255, 255);</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="NFA_Widget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>110</y>
      <width>550</width>
      <height>500</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);
border-color: rgb(85, 0, 255);</string>
    </property>
   </widget>
   <widget class="QWidget" name="DFA_Widget" native="true">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>110</y>
      <width>550</width>
      <height>500</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(255, 255, 255);</string>
    </property>
   </widget>
   <widget class="QPushButton" name="addEdgeBtn">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>820</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>MS Shell Dlg 2</family>
      <pointsize>12</pointsize>
      <weight>9</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 12pt &quot;MS Shell Dlg 2&quot;;
background-color: rgb(85, 0, 255);
color: rgb(255, 255, 255);
border-radius: 15px;</string>
    </property>
    <property name="text">
     <string>Add Edge</string>
    </property>
   </widget>
   <widget class="QPushButton" name="convertBtn">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>760</y>
      <width>271</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>MS Shell Dlg 2</family>
      <pointsize>14</pointsize>
      <weight>9</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 14pt &quot;MS Shell Dlg 2&quot;;
background-color: rgb(85, 0, 255);
color: rgb(255, 255, 255);
border-radius: 15px;</string>
    </property>
    <property name="text">
     <string>Convert NFA To DFA</string>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>640</y>
      <width>544</width>
      <height>160</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="gridLayout">
     <property name="horizontalSpacing">
      <number>0</number>
     </property>
     <item row="3" column="3" alignment="Qt::AlignHCenter">
      <widget class="QCheckBox" name="final2">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QLabel" name="label_5">
       <property name="font">
        <font>
         <family>MS Shell Dlg 2</family>
         <pointsize>13</pointsize>
         <weight>50</weight>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel{
	font: 13pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(85, 0, 255);
}</string>
       </property>
       <property name="text">
        <string>  Is Final?</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="toNodeTxt">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_3">
       <property name="font">
        <font>
         <family>MS Shell Dlg 2</family>
         <pointsize>13</pointsize>
         <weight>50</weight>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel{
	font: 13pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(85, 0, 255);
}</string>
       </property>
       <property name="text">
        <string>Transition Edge: </string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="fromNodeTxt">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="edgeLabelTxt">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_2">
       <property name="font">
        <font>
         <family>MS Shell Dlg 2</family>
         <pointsize>13</pointsize>
         <weight>50</weight>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel{
	font: 13pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(85, 0, 255);
}</string>
       </property>
       <property name="text">
        <string>To Node: </string>
       </property>
      </widget>
     </item>
     <item row="2" column="2" alignment="Qt::AlignHCenter">
      <widget class="QCheckBox" name="initial1">
       <property name="layoutDirection">
        <enum>Qt::LeftToRight</enum>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_4">
       <property name="font">
        <font>
         <family>MS Shell Dlg 2</family>
         <pointsize>13</pointsize>
         <weight>50</weight>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel{
	font: 13pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(85, 0, 255);
}</string>
       </property>
       <property name="text">
        <string>  Is Initial?</string>
       </property>
      </widget>
     </item>
     <item row="2" column="3" alignment="Qt::AlignHCenter">
      <widget class="QCheckBox" name="final1">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="3" column="2" alignment="Qt::AlignHCenter">
      <widget class="QCheckBox" name="initial2">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label">
       <property name="font">
        <font>
         <family>MS Shell Dlg 2</family>
         <pointsize>13</pointsize>
         <weight>50</weight>
         <italic>false</italic>
         <bold>false</bold>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">QLabel{
	font: 13pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(85, 0, 255);
}</string>
       </property>
       <property name="text">
        <string>From Node: </string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="clearGraphBtn">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>820</y>
      <width>151</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>MS Shell Dlg 2</family>
      <pointsize>12</pointsize>
      <weight>9</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(85, 0, 255);
font: 75 12pt &quot;MS Shell Dlg 2&quot;;
color: rgb(255, 255, 255);
border-radius: 15px;</string>
    </property>
    <property name="text">
     <string>Clear Graph</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>30</y>
      <width>291</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>30</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(85, 0, 255);
font: 30pt &quot;Gabriola&quot;;</string>
    </property>
    <property name="text">
     <string>Create Your NFA</string>
    </property>
   </widget>
   <widget class="QPushButton" name="generateGraphBtn">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>670</y>
      <width>271</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>MS Shell Dlg 2</family>
      <pointsize>14</pointsize>
      <weight>9</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">font: 75 14pt &quot;MS Shell Dlg 2&quot;;
background-color: rgb(85, 0, 255);
color: rgb(255, 255, 255);
border-radius: 15px;</string>
    </property>
    <property name="text">
     <string>Generate Random NFA</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>30</y>
      <width>291</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Gabriola</family>
      <pointsize>30</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">color: rgb(85, 0, 255);
font: 30pt &quot;Gabriola&quot;;</string>
    </property>
    <property name="text">
     <string>Convert To DFA</string>
    </property>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>-10</x>
      <y>100</y>
      <width>1141</width>
      <height>521</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">border: 2px rgb(85, 0, 255) ; 
border-style: inset;
</string>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="error_empty_fields">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>870</y>
      <width>221</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>MS Shell Dlg 2</family>
      <pointsize>11</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{

	font: 11pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(255, 0, 0);
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="star_from_node">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>700</y>
      <width>16</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>MS Shell Dlg 2</family>
      <pointsize>11</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	font: 11pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(255, 0, 0);
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="star_to_node">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>730</y>
      <width>16</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>MS Shell Dlg 2</family>
      <pointsize>11</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	font: 11pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(255, 0, 0);
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="star_transition_edge">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>760</y>
      <width>16</width>
      <height>27</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>MS Shell Dlg 2</family>
      <pointsize>11</pointsize>
      <weight>50</weight>
      <italic>false</italic>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QLabel{
	font: 11pt &quot;MS Shell Dlg 2&quot;;
	margin-right: 3px;
	color: rgb(255, 0, 0);
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <zorder>line</zorder>
   <zorder>NFA_Widget</zorder>
   <zorder>DFA_Widget</zorder>
   <zorder>addEdgeBtn</zorder>
   <zorder>convertBtn</zorder>
   <zorder>gridLayoutWidget</zorder>
   <zorder>clearGraphBtn</zorder>
   <zorder>label_6</zorder>
   <zorder>generateGraphBtn</zorder>
   <zorder>label_7</zorder>
   <zorder>error_empty_fields</zorder>
   <zorder>star_from_node</zorder>
   <zorder>star_to_node</zorder>
   <zorder>star_transition_edge</zorder>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
